src/test/java/org/broadinstitute/hellbender/engine/PipelineSupportIntegrationTest.java

```diff
@@ -0,0 +1,21 @@
+package org.broadinstitute.hellbender.engine;
+
+import org.broadinstitute.hellbender.GATKBaseTest;
+import org.broadinstitute.hellbender.utils.runtime.ProcessController;
+import org.broadinstitute.hellbender.utils.test.BaseTest;
+import org.testng.annotations.Test;
+
+import java.io.File;
+
+public class PipelineSupportIntegrationTest extends GATKBaseTest{
+
+    private static final String INPUT_BAM = toolsTestDir + "print_reads.sorted.chr1_1.bam";
+
+    @Test
+    // Asserting that Picard programs are able to pipe their output without errors
+    public void testPipeForPicardTools() {
+        File output = createTempFile("testOutput",".bam");
+        BaseTest.runProcess(ProcessController.getThreadLocal(), ("./gatk SortSam -I "+INPUT_BAM+" -O /dev/stdout -SO coordinate " +
+                "| ./gatk SetNmMdAndUqTags -I /dev/stdin -O "+ output.getAbsolutePath()+ " --CREATE_INDEX true -R "+hg19_chr1_1M_Reference).split(" "));
```

Comment:
Can you add an assert that after this pipeline runs, the output bam exists, and has an expected number of records in it?

Reply:
Sure, I considered this sufficient for the piping bug because it would exit with an error otherwise but for added security I will add proper asserts.